pkgs/by-name/ap/apx-gui/package.nix

```diff
@@ -0,0 +1,60 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, appstream-glib
+, desktop-file-utils
+, gobject-introspection
+, libadwaita
+, meson
+, ninja
+, pkg-config
+, python3
+, wrapGAppsHook4
+, apx
+, gnome-console
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "apx-gui";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner  = "Vanilla-OS";
+    repo   = "apx-gui";
+    rev    = "v${finalAttrs.version}";
+    hash = "sha256-orP5kAsoXX0zyDskeIPKKHNt5c757eUm9un4Ws6uFYA=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    appstream-glib
+    desktop-file-utils
+    gobject-introspection
+    meson
+    ninja
+    pkg-config
+    python3
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [
+    libadwaita
+  ];
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix PYTHONPATH : "${python3.pkgs.makePythonPath [ python3.pkgs.pygobject3 ]}"
+      --prefix PATH : "${lib.makeBinPath [ apx gnome-console ]}"
```

Comment:
Why gnome-console

Reply:
> Why gnome-console

Good question! Basically `apx-gui` is a frontend for `apx` so opening a terminal to run `apx` is one of the key functions of the program. Ideally this would be done in a DE agnostic way (is there a DE agnostic way to open a terminal on linux?). But the program hard-codes `kgx` i.e. `gnome-console`. This could be easily replaced with `substituteInPlace` but then what would we replace it with?  I thought bringing in `gnome-console` so that the program runs without expecting the user to have the right terminal program installed was the lesser of two evils. 

The alternative would be e.g. to call a script which goes through a list of default terminals and opens up the first one it finds, or something along those lines.